common/src/main/java/com/microsoft/identity/common/internal/util/PackageUtils.java

```diff
@@ -202,4 +208,31 @@ public static final X509Certificate getSelfSignedCert(final List<X509Certificate
 
         return selfSignedCert;
     }
+
+    public static String signatureVerificationAndThrow(final String packageName, final Context context,
+                                                       final Iterator<String> validPackageSignatures)
+            throws ClientException {
+        try {
+            // Read all the certificates associated with the package name. In higher version of
+            // android sdk, package manager will only returned the cert that is used to sign the
+            // APK. Even a cert is claimed to be issued by another certificates, sdk will return
+            // the signing cert. However, for the lower version of android, it will return all the
+            // certs in the chain. We need to verify that the cert chain is correctly chained up.
```

Comment:
This looks like copy-paste.  Can we research which version of android this is, so we know when to delete the logic?

Reply:
51443b0